tree/ntuple/v7/inc/ROOT/RFieldBase.hxx

```diff
@@ -363,13 +375,19 @@ protected:
    /// to a single column and has no read callback.
    void Read(NTupleSize_t globalIndex, void *to)
    {
-      if (fIsSimple)
-         return (void)fPrincipalColumn->Read(globalIndex, to);
+      if (fIsSimple) {
+         if (!fIsArtificial) {
+            fPrincipalColumn->Read(globalIndex, to);
+         }
+         return;
+      }
```

Comment:
@jblomer if I understand correctly, this is the place you were referring to? ie we want `fIsArtificial = true` to imply `fIsSimple = false` and then move conditions accordingly?

Reply:
Yes.